lib/configuration.js

```diff
@@ -31,58 +33,147 @@ var errors = therror.register({
+    merge(config, configSet);
+  }
+
+  /**
+   * Merge one or several configuration objects.
+   */
+  function mergeConfig() {
+    var args = Array.prototype.slice.call(arguments, 0);
+    args.unshift(config);
+    merge.apply(this, args);
+  }
 
-  var configuration = Object.create(nconf);
+  /**
+   * Merge the command-line arguments into the configuration.
+   *
+   * @param {String[]} whitelist
+   *    Optional list of environment variables to avoid including all of them.
+   */
+  function argv(whitelist) {
+    for (var key in yargs.argv) {
+      if (yargs.argv.hasOwnProperty(key)) {
+        if (!whitelist || whitelist.indexOf(key) >= 0) {
+          set(key, yargs.argv[key]);
+        }
+      }
+    }
+  }
 
   /**
-   * Load the environment variables into the configuration.
+   * Merge the environment variables into the configuration.
    * The keys from environment variables are transformed into camelCase to keep coherence with JSON format.
    *
    * @param {String[]} whitelist
    *    Optional list of environment variables to avoid including all of them.
    */
-  configuration.env = function(whitelist) {
-    var configEnv = {};
-    for (var key in process.env) {
-      if (process.env.hasOwnProperty(key)) {
-        var formattedKey = changeCase.camelCase(key);
-        if (!whitelist || whitelist.indexOf(formattedKey) >= 0) {
-          configEnv[formattedKey] = process.env[key];
+  function env(whitelist) {
+    for (var composedKey in process.env) {
+      if (process.env.hasOwnProperty(composedKey)) {
+        var keys = composedKey.split(delimiter);
+        var formattedComposedKey = keys.map(changeCase.camelCase).join(delimiter);
+        if (!whitelist || whitelist.indexOf(formattedComposedKey) >= 0) {
+          set(formattedComposedKey, process.env[composedKey]);
         }
       }
     }
-    nconf.add('configEnv', {type: 'literal', store: configEnv});
-  };
+  }
 
   /**
-   * Load the configuration from a JSON file, or the JSON files from a directory.
+   * Merge the configuration from a configuration file, or from the configuration files inside a directory.
+   * It supports either JSON files (with extension .json) with comments (via hjson module), or YML files.
    *
-   * @param {String} file
-   *    Path to the JSON file or directory.
+   * @param {String} filePath
+   *    Path to the configuration file or configuration directory.
    */
-  configuration.file = function(file) {
+  function file(filePath) {
 
-    var loadFile = function(configFile) {
-      if (path.extname(configFile) === '.json') {
-        nconf.file(configFile, {file: configFile, format: hjson});
-      } else {
-        throw errors.FILE_INVALID(configFile, 'Configuration file must have ".json" extension');
+    var loadFile = function(configFile, ignoreErrors) {
+      var fileExtension = path.extname(configFile);
+      if (fileExtension === '.json') {
+        var jsonConfig = hjson.parse(fs.readFileSync(configFile, 'utf8'));
```

Comment:
Does this mean that this lib supports the whole hjson spec? In that case I think it is worth to explicitly mention it.